carto/print.py

```diff
@@ -60,14 +61,15 @@ def __init__(self, username, map_id, api_key, width_cm, height_cm, zoom_level, b
         self.map_id = map_id
         self.filename = self.generate_filename()
         self.mode = mode
+        self.server_url = server_url + '/api/v1/map/static/named/{template}/{tile_size}/{tile_size}.png?zoom={zoom}&lat={lat}&lon={lon}&api_key={api_key}'
 
         self.validate_mode()
 
         if self.bounds is None and self.zoom is None:
             self.calculateBounds()
 
     def calculateBounds(self):
-        url = 'https://{username}.carto.com/api/v1/map/named/{tpl}?api_key={api_key}'.format(username=self.username, tpl=self.map_id, api_key=self.api_key)
+        url = self.server_url.format(username=self.username, tpl=self.map_id, api_key=self.api_key)
```

Comment:
I guess this won't work. The actual URL is a bit different than the template and the attributes are wrong. Maybe we can make `self.server_url = server_url` and build the complete URLs in each method.